### PR TITLE
Implement effects for transactions: https://www.stellar.org/developer…

### PR DESCRIPTION
**Pull request by david-pm**
https://www.stellar.org/developers/horizon/reference/endpoints/effects-for-transaction.html

seinfeld

### CODE REVIEW

File: client/src/endpoint/account.rs [hunk 1]
```diff
@@ -410,11 +410,11 @@ impl IntoRequest for Effects {
     }
 }
 
-// impl Cursor<Effect> for Effects {
-//     fn cursor(self, cursor: &str) -> Self {
-//         self.cursor(cursor)
-//     }
-// }
```
Comment (david-pm) [inline client/src/endpoint/account.rs]: this was commented out copy pasta... do we need it?

Comment (kbacha) [reply client/src/endpoint/account.rs]: Yes, the trait implementation is necessary for using the endpoint in an iterator.

File: client/src/endpoint/transaction.rs [hunk 2]
```diff
@@ -162,6 +162,188 @@ mod all_transactions_test {
...
+            limit: None,
+        }
+    }
+
+    /// Starts the page of results at a given cursor
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use stellar_client::sync::Client;
```
Comment (kbacha) [inline client/src/endpoint/transaction.rs]: We've just started moving the example that makes queries to horizon to just the struct definition itself. This way we provide a large example for clarity and reduce the number of http calls we make.

Here's an example of the new approach:
https://github.com/kbacha/stellar-sdk/blob/master/client/src/endpoint/account.rs#L456-L558

(note: we haven't converted the old examples yet)

Here's what the "new" way looks like in `cargo doc`:

image

File: client/src/endpoint/transaction.rs [hunk 3]
```diff
@@ -162,6 +162,188 @@ mod all_transactions_test {
...
+        let effects = Effects::new("abc123");
+        let req = effects
+            .into_request("https://horizon-testnet.stellar.org")
+            .unwrap();
+        assert_eq!(req.uri().path(), "/transactions/abc123/effects");
+        assert_eq!(req.uri().query(), None);
+    }
+
+    #[test]
+    fn it_can_make_a_actions_uri() {
```
Comment (kbacha) [inline client/src/endpoint/transaction.rs]: How is this different than the previous test?

**Comment by kbacha** [comment]
@david-pm Can you also add a `closes #123` to your description so that when this merges we automatically close the issue?

**Comment by david-pm** [comment]
gonna open another PR

**Pull request closed without merging**